connectors/email/src/main/java/io/camunda/connector/email/inbound/model/EmailListenerData.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.email.inbound.model;
+
+import io.camunda.connector.email.config.ImapConfig;
+import io.camunda.connector.generator.dsl.Property;
+import io.camunda.connector.generator.java.annotation.NestedProperties;
+import io.camunda.connector.generator.java.annotation.TemplateProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+
+public class EmailListenerData {
+
+  @Valid
+  @NestedProperties(addNestedPath = false)
+  private ImapConfig imapConfig;
+
+  @TemplateProperty(
+      label = "Folder to listen",
+      group = "listenerInfos",
+      id = "data.folderToListen",
+      description = "",
+      optional = true,
+      feel = Property.FeelMode.optional,
+      binding = @TemplateProperty.PropertyBinding(name = "data.folderToListen"))
+  private Object folderToListen;
```

Comment:
Lets go with a proper type instead of `Object`.

Reply:
The Object can really depends:

I can have 
```
{
  "field":"FROM",
  "value":"example@camunda.com"
}
```

or the example given previously, so the root object is not always the same